app/services/metrics_service.rb

```diff
@@ -7,12 +7,52 @@ def fetch(base_path:, from:, to:, metric:)
     client.get_json(url)
   end
 
+  def fetch_timeseries(base_path:, from:, to:, metric:)
+    url = time_series_request_url(base_path, from, to, metric)
+    data = client.get_json(url).to_hash
+    format_response(data, metric, from, to)
+  end
+
 private
 
+  def format_response(data, metric, from, to)
+    human_friendly_metric = human_friendly_metric(metric)
+    {
+      caption: "#{human_friendly_metric} from #{from.to_date} to #{to.to_date}",
+      chart_label: "#{human_friendly_metric} chart",
+      table_label: "#{human_friendly_metric} table",
+      chart_id: "#{metric}_#{from}-#{to}_chart",
+      table_id: "#{metric}_#{from}-#{to}_table",
+      keys: keys(data, metric),
+      rows: [
+        {
+          values: values(data, metric)
+        }
+      ]
+    }
+  end
+
+  def keys(data, metric)
+    dates = data[metric].map { |hash| hash['date'] }
+    dates.map { |date| date.last(5) }
+  end
+
+  def values(data, metric)
+    data[metric].map { |hash| hash['value'] }
+  end
+
+  def human_friendly_metric(metric)
```

Comment:
These private methods can't be tested directly. Another benefit of abstracting these as part of the refactor to presenter classes is that we'll be able to unit test them.